hypothesis-python/src/hypothesis/internal/conjecture/dfa/lstar.py

```diff
@@ -195,18 +345,25 @@ def seems_right(n):
             # that allows us to distinguish the state that we ended
             # up in from the state that we should have ended up in.
 
-            suffix = string[n + 1 :]
+            l1 = self.dfa.label(states[n + 1])
+            l2 = self.dfa.label(states[n]) + string[n : n + 1]
```

Comment:
I'd love to find some names that are less opaque than `l1` and `l2`, especially since they are handled quite differently.

(Indeed, it seems that `l2` only gets used in an assertion.)

Reply:
Brainstorming a bit here:

Since `states[n + 1]` is the state we plan to update (by adding a disambiguating experiment), we can think of it as the *target state*, which makes `l1` the *target label*.

Meanwhile, `l2` is a string that appears to lead to the target state in our current model, but should not. We can think of it as a *confused path* or *bogus path*.

(Specifically, `l2` is a path that extends label of `states[n]` with the single character `string[n]`.)

Furthermore, `l2` is only actually used in a subsequent assertion, which double-checks the findings of the prior loop which determined that `n` is where our model diverged from a correct understanding of input string.

Reply:
I've improved commenting and naming of this section a fair bit. What do you think now?

Reply:
I mostly like it, though I find it unfortunate that `current` and `correct` look almost identical at a distance.

I could keep bikeshedding, but I don’t know if it’s a reasonable use of your time, so I’d be willing to call this *fine* and move on if you want.

(Except for a couple of leftover references to `l1` and `l2` in a comment.)